config/collector.yaml

```diff
@@ -119,6 +132,15 @@ processors:
       - set(attributes["top_level_controller_type"], "CronJob") where resource.attributes["k8s.cronjob.name"] != nil
       - set(attributes["top_level_controller_name"], resource.attributes["k8s.cronjob.name"]) where resource.attributes["k8s.cronjob.name"] != nil
 
+# When sending telemetry to the GCP OTLP endpoint, the gcp.project_id resource attribute is required to be set to your project ID.
+  resource/gcp_project_id:
+    attributes:
+    - key: gcp.project_id
+      # MAKE SURE YOU REPLACE THIS WITH YOUR PROJECT ID
```

Comment:
Instead of making the user replace the value here, would it make sense to use collector support for ENV variables substitution?

https://opentelemetry.io/docs/collector/configuration/#environment-variables

It looks like it should be valid to specify something like:

```
 value: ${env:GOOGLE_CLOUD_PROJECT}
``` 

... and to have this get expanded at runtime by the collector, assuming that the variable is -- in fact -- defined in the environment.

Reply:
I'm not sure about this. Isn't this syntax only valid for use with downward API, which means pods should have access to the value of this variable?

Reply:
I tested this and I think this method won't work here. My understanding is that this value needs to be available in the pod spec at runtime. 
You can see this in environment variables like `MY_POD_IP` used in these manifests which are set using the downward API in the pod spec. I don't think PROJECT_ID would be available unless explicitly mounted/set in the pod spec, at which point, it kinda defeats the purpose of providing the convenience automatically reading the environment variables.

Reply:
Seems reasonable.

It looks like this would require wrapping the collector invocation with a script to set these environment variables from the output of querying the Google Metadata server.

I agree that this would probably make things unnecessarily complicated for sample code.

It might make sense, though, to leave some comments hinting at this to make it easier for someone wanting to deploy the same configuration to multiple destinations without needing to fork the configuration for each deployment project.

Reply:
The README instructions mention the use of `envsubst` to get around this. I can add comments here indicating the same.